src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -252,8 +252,10 @@ moduleToJs (Module coms mn _ imps exps foreigns decls) foreign_ =
   iife v exprs = AST.App Nothing (AST.Function Nothing Nothing [] (AST.Block Nothing $ exprs ++ [AST.Return Nothing $ AST.Var Nothing v])) []
 
   literalToValueJS :: Literal (Expr Ann) -> m AST
-  literalToValueJS (NumericLiteral (Left i)) = return $ AST.NumericLiteral Nothing (Left i)
-  literalToValueJS (NumericLiteral (Right n)) = return $ AST.NumericLiteral Nothing (Right n)
+  literalToValueJS (NumericLiteral (LitInt i)) = return $ AST.NumericLiteral Nothing (Left i)
+  literalToValueJS (NumericLiteral (LitNumber n)) = return $ AST.NumericLiteral Nothing (Right n)
+  -- TODO: propagate the natural-ness of the unsigned integer to the AST
```

Comment:
Clearly, a WIP :slightly_smiling_face: Curious for feedback on continuing to integrate into the compiler.

Reply:
This seems sensible to me so far. I suppose the next step would be to modify the `AST` type in `Language.PureScript.CoreImp.AST` to use `NumericLiteral` instead of `Either Integer Double`?